openquake/server/views.py

```diff
@@ -777,7 +805,12 @@ def aristotle_run(request):
     (usgs_id, lon, lat, dep, mag, rake, dip, strike, maximum_distance, trt,
      truncation_level, number_of_ground_motion_fields,
      asset_hazard_distance, ses_seed) = res
-    rupture_file = None # to be set from the upload form
+    # NOTE: in the current approach, the rupture file is used to pre-populate
+    # the editable rupture fields, then it is not used below anymore. Another
+    # approach could be to disable the editable rupture fields if a rupture
+    # model xml is provided, then to pass the rupture file to
+    # get_aristotle_allparams below
+    rupture_file = None
```

Comment:
This is WRONG: the whole point of using a rupture_file is to pass the geometry to the calculator and if you pass rupture_file = None all is lost.

Reply:
Ok, then I will use the second approach described in the code. It will imply that the presence of the rupture file will automatically disable the form fields displaying the rupture details, so they will be visible but not editable. If we also want to remove the duplication pointed out in your other comment, the rupture fields can not be filled at all, and they must remain disabled and empty.